application/helpers/frontend_helper.php

```diff
@@ -1886,13 +1889,39 @@ function doAssessment($surveyid, $returndataonly=false)
                             if ($usresult)
                             {
                                 $usrow = $usresult->read();
-                                $fieldmap[$field['fieldname']]['assessment_value']=$usrow['assessment_value'];
-                                $total=$total+$usrow['assessment_value'];
+                                $assessmentValue = $usrow['assessment_value'];
                             }
                         }
                     }
                     $groups[]=$field['gid'];
                 }
+
+                // If this is a question (and not a survey field, like ID), save asessment value
+                if (!empty($field['title']))
+                {
+                    /**
+                     * Allow Plugin to update assessment value
+                     */
+                    // Prepare Event Info
+                    $event = new PluginEvent('afterSurveyQuestionAssessment');
+                    $event->set('surveyId', $surveyid);
+                    $event->set('question', $field);
```

Comment:
> Partly agree. I am more into saying some basic information shall be shared among plugins and that there can be specific info for each plugin.

Nothing must be specific for plugin, only for event. And here SGQA/qid/surveyid seems to be widely used after in any plugin.

We already have : surveyid or survey for survey id :/ . I added gid because seems to be a common var for beforeQuestionRender and some other (i don't need it specially , i can play without) , but try to get the middle betwwen : needed and lot used information and more information than needed. .